ddcommon/src/lib.rs

```diff
@@ -32,11 +32,22 @@ pub type HttpClient = hyper::Client<connector::Connector, hyper::Body>;
 pub type HttpResponse = hyper::Response<hyper::Body>;
 pub type HttpRequestBuilder = hyper::http::request::Builder;
 
-#[derive(Default, Clone, PartialEq, Eq, Hash, Debug, Serialize, Deserialize)]
+#[derive(Clone, PartialEq, Eq, Hash, Debug, Serialize, Deserialize)]
 pub struct Endpoint {
     #[serde(serialize_with = "serialize_uri", deserialize_with = "deserialize_uri")]
     pub url: hyper::Uri,
     pub api_key: Option<Cow<'static, str>>,
+    pub timeout: u64,
```

Comment:
Is there a particular reason to not have this as a `Duration`?

Reply:
The aim was to reduce a little bit of memory. Since `Endpoint` is embedded in each `SendData` structurure and, therefore, the `TraceFlusher` holds a collection of `SendData` objects I thought it would be better just to hold 8 bytes rather than 16 per `SendData` instance. I even thought of just holding a u16 (probably a timeout of 65535 would be enough for our purposes) but since all milliseconds values inside std::time were expressed as a u64 I left it that way.

Reply:
At the very least, this should document the unit, e.g. `timeout_ms`.  I'd argue for the `Duration` type as more self-documenting: do we have numbers for how many bytes this actually saves?

Reply:
@danielsn: Yeah, that'll make things clearer. Didn't have time to do benchnmark it. 
@ekump: `SendData` instances are eventually merged together (up to a maximum payload size) when the `TraceFlusher` tries flush them out so not every timeout is converted to a `Duration` that's why it seemed a waste of space.